pkg/configauditreport/controller/controller.go

```diff
@@ -357,38 +409,92 @@ func (r *ResourceController) reconcileConfig(kind kube.Kind) reconcile.Func {
 		if err != nil {
 			return ctrl.Result{}, fmt.Errorf("parsing label selector: %w", err)
 		}
-
-		var reportList v1alpha1.ConfigAuditReportList
-		err = r.Client.List(ctx, &reportList,
-			client.Limit(r.Config.BatchDeleteLimit+1),
-			client.MatchingLabelsSelector{Selector: labelSelector})
+		configRequeueAfter, err := r.deleteReports(ctx, labelSelector, &v1alpha1.ConfigAuditReportList{})
 		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("listing reports: %w", err)
+			return ctrl.Result{}, err
 		}
+		rbacRequeueAfter, err := r.deleteReports(ctx, labelSelector, &v1alpha1.RbacAssessmentReportList{})
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if configRequeueAfter || rbacRequeueAfter {
+			return ctrl.Result{RequeueAfter: r.Config.BatchDeleteDelay}, nil
+		}
+		return ctrl.Result{}, nil
+	}
+}
 
-		log.V(1).Info("Listing ConfigAuditReports",
-			"reportsCount", len(reportList.Items),
-			"batchDeleteLimit", r.Config.BatchDeleteLimit,
-			"labelSelector", labelSelector.String())
-
-		for i := 0; i < ext.MinInt(r.Config.BatchDeleteLimit, len(reportList.Items)); i++ {
-			report := reportList.Items[i]
-			log.V(1).Info("Deleting ConfigAuditReport", "report", report.Namespace+"/"+report.Name)
-			err := r.Client.Delete(ctx, &report)
+func (r *ResourceController) deleteReports(ctx context.Context, labelSelector labels.Selector, reportList client.ObjectList) (bool, error) {
```

Comment:
generics will not relate here :)  the problem is that we are luck of common interface for reports.
generic is just to force but with out common interface it will not be easy to reduce code.
common interface.
will try to do common interface on different PR , however it will be also ugly , maybe less ugly